vignettes/scoringutils.Rmd

```diff
@@ -79,10 +79,10 @@ The output of `validate()` can later be directly used as input to `score()` (oth
 
 ## Showing available forecasts
 
-The function `available_forecasts()` may also be helpful to determine where forecasts are available. Using the `by` argument you can specify the level of summary. For example, to see how many forecasts there are per model and target_type, we can run
+The function `get_forecast_counts()` may also be helpful to determine where forecasts are available. Using the `by` argument you can specify the level of summary. For example, to see how many forecasts there are per model and target_type, we can run
 
 ```{r}
-available_forecasts(example_quantile, by = c("model", "target_type"))
+get_forecast_counts(example_quantile, by = c("model", "target_type"))
```

Comment:
Ultimately the result should be exactly the same, regardless of whether you run `as_forecast()` on the data - if I'm not mistaken. 
At the moment `get_forecast_counts()` does some shenanigans with a `forecast_unit` attribute, but eventually I think it should just call `get_forecast_unit()` internally

Reply:
my point is that if you demonstrated this workflow with that flow you could avoid setting `by` here and also demonstrate our proposed workflow.

Reply:
Still not sure I understand. If you don't set `by`, then `by` will just be the forecast unit and then you end up with a big data.table that has counts either 0 or 1 - regardless of whether you call `as_forecast()` before or not. 

<img width="816" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/7e9e75e7-fc6d-4d54-8beb-199e37cfd414">

But of course we can update the example to call `as_forecast()` before (or `validate()` now)

Reply:
I meant explicitly setting the forecast unit. I also think this could be part of the `as_forecast` workflow as this would be quite clean but that is a separate issue.